test/extensions/filters/http/oauth2/oauth_test.cc

```diff
@@ -492,6 +483,49 @@ TEST_F(OAuth2ClientTest, NoCluster) {
   EXPECT_EQ(0, callbacks_.size());
 }
 
+TEST_F(OAuth2ClientTest, RequestAccessTokenRetryPolicy) {
+  envoy::config::core::v3::HttpUri uri;
+  uri.set_cluster("auth");
+  uri.set_uri("auth.com/oauth/token");
+  uri.mutable_timeout()->set_seconds(1);
+
+  envoy::config::route::v3::RetryPolicy retry_policy;
+  retry_policy.set_retry_on("5xx,reset");
+  retry_policy.mutable_retry_back_off()->mutable_base_interval()->set_seconds(1);
+  retry_policy.mutable_retry_back_off()->mutable_max_interval()->set_seconds(10);
+  retry_policy.mutable_num_retries()->set_value(5);
+
+  client_ = std::make_shared<OAuth2ClientImpl>(cm_, uri, retry_policy, 2000s);
+
+  EXPECT_CALL(cm_.thread_local_cluster_.async_client_, send_(_, _, _))
+      .WillRepeatedly(Invoke(
```

Comment:
Why `WillRepeatedly` is used? I think the send_ should be called only once. The retry is handled by the Router internally.